include/pmacc/random/distributions/uniform/Uniform_double.hpp

```diff
@@ -0,0 +1,177 @@
+
+namespace pmacc
+{
+namespace random
+{
+namespace distributions
+{
+namespace detail
+{
+
+    /** Returns a random double value uniformly distributed in (0,1]
+     *
+     * The smallest created value is `2^-65` (~ `2.710505431213761*10^-20`)
+     */
+    template<class T_RNGMethod>
+    class Uniform<
+        uniform::ExcludeZero<double>,
+        T_RNGMethod,
+        void
+    >
+    {
+    public:
+        typedef T_RNGMethod RNGMethod;
+        typedef typename RNGMethod::StateType StateType;
+        typedef double result_type;
+
+        template< typename T_Acc >
+        DINLINE double
+        operator()(
+            T_Acc const & acc,
+            StateType& state
+        ) const
+        {
+            double const value2pow64Inv = 5.421010862427522e-20;
+            uint64_t const random = RNGMethod().get64Bits(
+                acc,
+                state
+            );
+            return static_cast< double >( random ) * value2pow64Inv +
+                ( value2pow64Inv / 2.0 );
+        }
+    };
+
+    /** Returns a random double value uniformly distributed in [0,1)
+     *
+     * Swap the value one to zero (creates a small error in uniform distribution)
+     */
+    template<class T_RNGMethod>
+    class Uniform<
+        uniform::ExcludeOne< double >::SwapOneToZero,
+        T_RNGMethod,
+        void
+    >
+    {
+    public:
+        typedef T_RNGMethod RNGMethod;
+        typedef typename RNGMethod::StateType StateType;
+        typedef double result_type;
+
+        template< typename T_Acc >
+        DINLINE double
+        operator()(
+            T_Acc const & acc,
+            StateType& state
+        ) const
+        {
+            double const randomValue =
+                pmacc::random::distributions::Uniform<
+                    uniform::ExcludeZero< double >,
+                    RNGMethod
+            >()(acc, state);
+            return randomValue == 1.0 ? 0.0 : randomValue;
+        }
+    };
+
+    /** Returns a random double value uniformly distributed in [0,1)
+     *
+     * Reduce the random range to `2^53`.
+     * Uses a uniform distance of `2^-53` (`epsilon/2`) between each possible
+     * random number.
+     */
+    template<class T_RNGMethod>
+    class Uniform<
+        uniform::ExcludeOne< double >::Reduced,
+        T_RNGMethod,
+        void
+    >
+    {
+    public:
+        typedef T_RNGMethod RNGMethod;
+        typedef typename RNGMethod::StateType StateType;
+        typedef double result_type;
+
+        template< typename T_Acc >
+        DINLINE double
+        operator()(
+            T_Acc const & acc,
+            StateType& state
+        ) const
+        {
+            double const value2pow53Inv = 1.1102230246251565e-16;
+            double const randomValue53Bit = RNGMethod().get64Bits( acc, state ) >> 11;
+            return randomValue53Bit * value2pow53Inv;
+        }
+    };
+
+    /** Returns a random double value uniformly distributed in [0,1)
```

Comment:
QUESTION:
For what do we need this function with a **uniform** distribution. This brute force approach works great for arbitrary distributions.

Reply:
It makes only sense for uniform distribution because the ranges for the normal distribution differ.

Reply:
Here, "normal" means not normal in the sense of Gaussian probability distribution, but in the sense of standard [0,1) or (nvidia) (0,1] implementation - and thus means (0,1). 
Thanks for fixing the comment.